src/middleware/preAuth.middleware.ts

```diff
@@ -16,7 +17,7 @@ export class PreauthMiddleware implements NestMiddleware {
         .auth()
         .verifyIdToken(token.replace('Bearer ', ''))
         .then(async (decodedToken) => {
-          const { email, uid } = decodedToken;
+          const { email, uid, picture } = decodedToken;
```

Comment:
add display name also.
And store it in firstName field with spliting through space.